packages/@aws-cdk/aws-apprunner/lib/service.ts

```diff
@@ -798,25 +809,44 @@ export class Service extends cdk.Resource {
 
   }
   private renderCodeConfigurationValues(props: CodeConfigurationValues): any {
+    this.environment = props.environment;
     return {
-      ...props,
+      port: props.port,
+      buildCommand: props.buildCommand,
       runtime: props.runtime.name,
+      runtimeEnvironmentVariables: this.renderEnvironmentVariables(),
+      startCommand: props.startCommand,
     };
   }
   private renderImageRepository(): any {
     const repo = this.source.imageRepository!;
+    this.environment = repo.imageConfiguration?.environment;
     if (repo.imageConfiguration?.port) {
       // convert port from type number to string
       return Object.assign(repo, {
         imageConfiguration: {
           port: repo.imageConfiguration.port.toString(),
+          startCommand: repo.imageConfiguration.startCommand,
+          runtimeEnvironmentVariables: this.renderEnvironmentVariables(),
```

Comment:
What happens to these values when `port` is not provided?

Reply:
Thanks for pointing out this stupid mistake. Should all fixed. Please take a look again.  @corymhall

Reply:
Hey @neilkuan are you able to take a peek at this? 😸